tests/payjoin.rs

```diff
@@ -42,18 +41,26 @@ async fn payjoin() -> Result<()> {
     info!("Address: {}", wallet.address);
 
     info!("Initiating PayJoin using BIP-21");
-    let address = env::var("MAIN_VAULT_ADDRESS")?;
+    let address = wallet.address;
     let destination = format!("bitcoin:{address}?pj=https://testnet.demo.btcpayserver.org/BTC/pj");
     let amount = 1000;
 
-    send_sats(
+    match send_sats(
         &vault.btc_descriptor_xprv,
         &vault.btc_change_descriptor_xprv,
         &destination,
         amount,
         Some(1.1),
     )
-    .await?;
+    .await
+    {
+        Ok(_) => {
+            panic!("Unexpected");
+        }
+        Err(e) => {
+            assert_eq!(e.to_string(), "couldn't decode PSBT");
```

Comment:
Still getting couldn't decode PSBT on testnet? That means the PSBT is malformed or de/encoded incorrectly

Reply:
This is actually the same error returned, even if there's different errors underneath. In the console, this actually returns a richer error message, but it isn't added to the Rust Error result type. I believe this is an issue in the upstream implementation.

Reply:
Ok, yes, I was able to create [a PayJoin on testnet](https://blockstream.info/testnet/tx/b351ed0685b8db179a12a3ee001115743078a04d6b66c830d3a929f8b1d7c441) from the test by changing the destination string to a bip21 from testnet.demo.btcpayserver.org via manual intervention.

It works!

There are two remaining (non-blocking IMO) problems with the test

1. regtest tries to hit the btcpayserver testnet server which won't find inputs
2. the bip21 is hard coded to `wallet.address` with no corresponding invoice at the btcpayserver. Those invoices expire after 15 minutes.

Reply:
I think those are both non-blocking issues as well, though I would still like to be able to make sure the expected error message is returned from the payjoin library. It'd increase the utility of the test a bit more than it is now, and shouldn't be too much extra work if another payjoin release can be made. What do you think?

Reply:
payjoin = "0.5.1-alpha" has been released for you!

Reply:
Excellent! I'll get it updated right away!